packages/store/dbstore.go

```diff
@@ -0,0 +1,207 @@
+	Host     string
+	Port     int
+	User     string
+	Password string
+	DBName   string
+	DB       *sql.DB
+}
+
+func (p *PostgreSQL) Listen() chan<- set.Token {
+	// listen to all tokens passed into the channel
+	res := make(chan set.Token)
+	go func() {
+		for input := range res {
+			go func() {
+				var port int
+				var attackType string
+				timestamp := int64(input.TOE)
+				timeObj := time.Unix(timestamp, 0)
+				if input.Events[honeypot.PortEventID] != nil {
+					port = input.Events[honeypot.PortEventID]["port"].(int)
+					attackType = "Port Scanning"
+				}
+				if input.Events[honeypot.LoginEventID] != nil {
+					attackType = "Login Attempt"
+					port = input.Events[honeypot.LoginEventID]["port"].(int)
+					username := input.Events[honeypot.LoginEventID]["username"].(string)
+					password := input.Events[honeypot.LoginEventID]["password"].(string)
+					service := input.Events[honeypot.LoginEventID]["service"].(string)
+					defer p.loginattInsert(input.JTI, service, username, password)
+				}
+				if input.Events[honeypot.HTTPEventID] != nil {
+					attackType = "HTTP Request"
+					method := input.Events[honeypot.HTTPEventID]["method"].(string)
+					path := input.Events[honeypot.HTTPEventID]["path"].(string)
+					port = input.Events[honeypot.HTTPEventID]["port"].(int)
+					acceptLanguage := input.Events[honeypot.HTTPEventID]["accept-lang"].(string)
+					useragent := input.Events[honeypot.HTTPEventID]["user-agent"].([]string)
+					if method == "POST" || method == "PUT" || method == "PATCH" {
+						payloadSize := input.Events[honeypot.HTTPEventID]["bodysize"].(int)
+						maxSize := int64(100 * 1024 * 1024)
+						//Payload size should be greater than 100MB
+						if payloadSize < int(maxSize) {
+							attackID := input.JTI
+							payload := input.Events[honeypot.HTTPEventID]["body"].(string)
+							savePayload(attackID, payload)
+						} else {
+							slog.Info("Payload size is greater than 100MB")
+						}
+						contentType := input.Events[honeypot.HTTPEventID]["content-type"].(string)
+						defer p.bodyInsert(input.JTI, method, contentType, strconv.Itoa(payloadSize)+" bytes")
+					}
+					defer p.httpInsert(input.JTI, method, path, acceptLanguage, useragent)
+				}
+				p.attackInsert(input.JTI, timeObj, port, input.SUB, input.COUNTRY, attackType)
+			}()
+		}
+	}()
+	return res
+}
+
+// Insert the attack into the database and sanitize the input by using prepared statements
+func (p *PostgreSQL) attackInsert(attackID string, time time.Time, port int, ip string, country string, attackType string) {
+	_, err := p.DB.Exec(`
+	INSERT INTO attack_log (Attack_id, Time_Of_Event,Port_Nr,IP_Address,Country,Attack_Type)
+	VALUES ($1, $2, $3, $4,$5, $6);
+	`, attackID, time, port, ip, country, attackType)
+	if err != nil {
+		log.Println("Error inserting into the database attack_log", err)
+	}
+}
+
+func (p *PostgreSQL) loginattInsert(attackID string, service string, username string, password string) {
```

Comment:
```suggestion
func (p *PostgreSQL) loginAttemptInsert(attackID string, service string, username string, password string) {
```

I think we should use the most speaking variable names possible